### PR TITLE
feat(faucet): don't build faucet if custom branch is not set

### PR DESCRIPTION
**Pull request by RolandSherwin**

### CODE REVIEW

File: src/tests/provision_faucet.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ async fn should_run_ansible_against_the_remaining_nodes_with_a_custom_binary() -
             eq(PathBuf::from("inventory").join(".beta_genesis_inventory_digital_ocean.yml")),
             eq("root".to_string()),
             eq(Some(
-                "{ \"provider\": \"digital-ocean\", \"testnet_name\": \"beta\", \"genesis_multiaddr\": \"/ip4/10.0.0.10/tcp/12000/p2p/12D3KooWLvmkUDQRthtZv9CrzozRLk9ZVEHXgmx6UxVMiho5aded\", \"branch\": \"custom_branch\", \"org\": \"maidsafe\" }".to_string()
+                "{ \"provider\": \"digital-ocean\", \"testnet_name\": \"beta\", \"genesis_multiaddr\": \"/ip4/10.0.0.10/tcp/12000/p2p/12D3KooWLvmkUDQRthtZv9CrzozRLk9ZVEHXgmx6UxVMiho5aded\", \"branch\": \"custom_branch\", \"org\": \"maidsafe\", \"faucet_archive_url\": \"https://sn-node.s3.eu-west-2.amazonaws.com/maidsafe/custom_branch/faucet-beta-x86_64-unknown-linux-musl.tar.gz\" }".to_string(),
```
Comment (joshuef) [inline src/tests/provision_faucet.rs:70]: why is there `beta` in here? the branch is `custom_branch`, no?

Comment (RolandSherwin) [reply src/tests/provision_faucet.rs:70]: `beta` is the testnet name. A binary that is built by the build-vm is stored with the following name:
```
bin_archive_filename: "{{ bin_name }}-{{ testnet_name }}-x86_64-unknown-linux-musl.tar.gz"
```

Comment (RolandSherwin) [reply src/tests/provision_faucet.rs:70]: To this following s3 location `object: /{{ org }}/{{ branch }}/{{ bin_archive_filename }}`

**Comment by RolandSherwin** [comment]
Superseded by #54

**Pull request closed without merging**